backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -774,4 +781,41 @@ IGitMacheteRepositorySnapshot discoverGitMacheteRepository() throws GitMacheteEx
     }
 
   }
+
+  @Getter
+  private static class GitMacheteNonRootBranchCreationResult {
+    private final List<GitMacheteNonRootBranch> createdBranches;
+    private final List<String> notCreatedBranchNames;
+
+    private GitMacheteNonRootBranchCreationResult() {
+      createdBranches = List.empty();
+      notCreatedBranchNames = List.empty();
+    }
+
+    public static GitMacheteNonRootBranchCreationResult empty() {
+      return new GitMacheteNonRootBranchCreationResult();
+    }
+
+    GitMacheteNonRootBranchCreationResult(GitMacheteNonRootBranchCreationResult prevResult, String notCreatedBranch) {
+      createdBranches = prevResult.getCreatedBranches();
+      notCreatedBranchNames = prevResult.getNotCreatedBranchNames().append(notCreatedBranch);
+    }
+
+    GitMacheteNonRootBranchCreationResult(GitMacheteNonRootBranch createdBranch, List<String> notCreatedBranchNames) {
+      createdBranches = List.of(createdBranch);
+      this.notCreatedBranchNames = notCreatedBranchNames;
+    }
+
+    GitMacheteNonRootBranchCreationResult(GitMacheteNonRootBranchCreationResult prevResult1,
```

Comment:
And TBH this one would looks to me like a static method `merge` with the same params